### PR TITLE
Basic support for dwds in incremental compiler

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

As is this is not intended to be landed, but gives us a good idea about the integration for dwds/incremental compiler

fyi @grouma

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart [hunk 1]
```diff
@@ -530,6 +532,13 @@ class _ExperimentalResidentWebRunner extends ResidentWebRunner {
       });
       _wipConnection = await chromeTab.connect();
     }
+    final WebDevFS webDevFS = device.devFS as WebDevFS;
+    final Dwds dwds = webDevFS.dwds;
+    dwds.connectedApps.listen((AppConnection appConnection) async {
+      appConnection.runMain();
+      _debugConnection = await dwds.debugConnection(appConnection);
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart]: To do things with the compile expression support, you'd want to grab the vmservice here and register a new method that forwards to the underling resident compiler

File: packages/flutter_tools/lib/src/web/devfs_web.dart [hunk 2]
```diff
@@ -317,11 +324,12 @@ class WebDevFS implements DevFS {
         mapperUrl: _filePathToUriFragment(stackTraceMapper.path),
         entrypoint: '${_filePathToUriFragment(mainPath)}.js',
       ));
-      _webAssetServer.writeFile('/main_module.js', generateMainModule(
+      _webAssetServer.writeFile('/main_module.bootstrap.js', generateMainModule(
         entrypoint: '${_filePathToUriFragment(mainPath)}.js',
       ));
       _webAssetServer.writeFile('/dart_sdk.js', dartSdk.readAsStringSync());
       _webAssetServer.writeFile('/dart_sdk.js.map', dartSdkSourcemap.readAsStringSync());
+      _webAssetServer.writeFile('/basic.digests', '{}'); // write fake digests.
```
Comment (grouma) [inline packages/flutter_tools/lib/src/web/devfs_web.dart]: Just confirming. This is only necessary if the module strategy is `requireJS`. Is that the case here?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/web/devfs_web.dart]: Yeah, we're still using requireJS

**Comment by jonahwilliams** [comment]
Closing in favor of another PR

**Pull request closed without merging**